src/data/polynomial/ring_division.lean

```diff
@@ -881,6 +843,76 @@ by { convert C_leading_coeff_mul_prod_multiset_X_sub_C hroots, rw [hp.leading_co
 
 end comm_ring
 
+section
+variables {A B : Type*} [comm_ring A] [comm_ring B]
+
+lemma le_root_multiplicity_map {p : A[X]} {f : A →+* B} (hmap : map f p ≠ 0) (a : A) :
+  root_multiplicity a p ≤ root_multiplicity (f a) (p.map f) :=
+begin
+  rw [le_root_multiplicity_iff hmap],
+  refine trans _ ((map_ring_hom f).map_dvd (pow_root_multiplicity_dvd p a)),
+  rw [map_pow, map_sub, coe_map_ring_hom, map_X, map_C],
+end
+
+lemma eq_root_multiplicity_map {p : A[X]} {f : A →+* B} (hf : function.injective f)
+  (a : A) : root_multiplicity a p = root_multiplicity (f a) (p.map f) :=
+begin
+  by_cases hp0 : p = 0, { simp only [hp0, root_multiplicity_zero, polynomial.map_zero], },
+  apply le_antisymm (le_root_multiplicity_map ((polynomial.map_ne_zero_iff hf).mpr hp0) a),
+  rw [le_root_multiplicity_iff hp0, ← map_dvd_map f hf ((monic_X_sub_C a).pow _),
+    polynomial.map_pow, polynomial.map_sub, map_X, map_C],
+  apply pow_root_multiplicity_dvd,
+end
+
+lemma count_map_roots [is_domain A] {p : A[X]} {f : A →+* B}
+  (hmap : map f p ≠ 0) (b : B) :
+  (p.roots.map f).count b ≤ root_multiplicity b (p.map f) :=
+begin
+  rw [le_root_multiplicity_iff hmap, ← multiset.prod_repeat, ← multiset.map_repeat (λ a, X - C a)],
+  rw ← multiset.filter_eq,
+  refine (multiset.prod_dvd_prod_of_le $ multiset.map_le_map $ multiset.filter_le _ _).trans _,
+  convert polynomial.map_dvd _ p.prod_multiset_X_sub_C_dvd,
+  simp only [polynomial.map_multiset_prod, multiset.map_map],
+  congr, ext1,
+  simp only [function.comp_app, polynomial.map_sub, map_X, map_C],
+end
+
+lemma count_map_roots_of_injective [is_domain A] (p : A[X]) {f : A →+* B}
+  (hf : function.injective f) (b : B) :
+  (p.roots.map f).count b ≤ root_multiplicity b (p.map f) :=
+begin
+  by_cases hp0 : p = 0,
+  { simp only [hp0, roots_zero, multiset.map_zero,
+      multiset.count_zero, polynomial.map_zero, root_multiplicity_zero] },
+  { exact count_map_roots ((polynomial.map_ne_zero_iff hf).mpr hp0) b },
+end
+
+lemma map_roots_le [is_domain A] [is_domain B] {p : A[X]} {f : A →+* B} (h : p.map f ≠ 0) :
+  p.roots.map f ≤ (p.map f).roots :=
+multiset.le_iff_count.2 $ λ b, by { rw count_roots, apply count_map_roots h }
+
+lemma map_roots_le_of_injective [is_domain A] [is_domain B] (p : A[X])
+  {f : A →+* B} (hf : function.injective f) :
+  p.roots.map f ≤ (p.map f).roots :=
+begin
+  by_cases hp0 : p = 0, { simp only [hp0, roots_zero, multiset.map_zero, polynomial.map_zero], },
+  exact map_roots_le ((polynomial.map_ne_zero_iff hf).mpr hp0),
+end
+
+lemma card_roots_le_map [is_domain A] [is_domain B] (p : A[X]) {f : A →+* B} (h : p.map f ≠ 0) :
+  p.roots.card ≤ (p.map f).roots.card :=
+by { rw ← p.roots.card_map f, exact multiset.card_le_of_le (map_roots_le h) }
+
+lemma roots_map_of_injective_card_eq_total_degree [is_domain A] [is_domain B] {p : A[X]}
```

Comment:
```suggestion
lemma roots_map_of_injective_of_card_eq_nat_degree [is_domain A] [is_domain B] {p : A[X]}
```

Reply:
This is used in [splitting_field](https://github.com/leanprover-community/mathlib/blob/a0ae2165d015aef83609c5d48f1e58320697df82/src/field_theory/splitting_field.lean#L276) but I agree this PR is a good opportunity to correct the weird name.